plugins/inventory/iocage.py

```diff
@@ -0,0 +1,253 @@
+        cache_key = self.get_cache_key(path)
+
+        user_cache_setting = self.get_option('cache')
+        attempt_to_read_cache = user_cache_setting and cache
+        cache_needs_update = user_cache_setting and not cache
+
+        if attempt_to_read_cache:
+            try:
+                results = self._cache[cache_key]
+            except KeyError:
+                cache_needs_update = True
+        if not attempt_to_read_cache or cache_needs_update:
+            results = self.get_inventory(path)
+        if cache_needs_update:
+            self._cache[cache_key] = results
+
+        self.populate(results)
+
+    def get_inventory(self, path):
+        host = self.get_option('host')
+        env = self.get_option('env')
+        get_properties = self.get_option('get_properties')
+
+        cmd = []
+        if host != 'localhost':
+            user = self.get_option('user')
+            cmd.append("ssh")
+            cmd.append(f"{user}@{host}")
+        if env:
+            cmd.extend(env)
+        cmd.append(self.IOCAGE)
+
+        cmd_list = cmd.copy()
+        cmd_list.append('list')
+        cmd_list.append('--header')
+        cmd_list.append('--long')
+        try:
+            p = Popen(cmd_list, stdout=PIPE, stderr=PIPE)
+            stdout, stderr = p.communicate()
+            if p.returncode != 0:
+                raise AnsibleError('Failed to run cmd=%s, rc=%s, stderr=%s' %
+                                   (cmd_list, p.returncode, to_native(stderr)))
+
+            try:
+                t_stdout = to_text(stdout, errors='surrogate_or_strict')
+            except UnicodeError as e:
+                raise AnsibleError('Invalid (non unicode) input returned: %s' % to_native(e))
+
+            iocage_data = [x.split() for x in t_stdout.splitlines()]
+            results = {'_meta': {'hostvars': {}}}
+
+            for jail in iocage_data:
+                iocage_name = jail[1]
+                results['_meta']['hostvars'][iocage_name] = {}
+                results['_meta']['hostvars'][iocage_name]['iocage_jid'] = jail[0]
+                results['_meta']['hostvars'][iocage_name]['iocage_boot'] = jail[2]
+                results['_meta']['hostvars'][iocage_name]['iocage_state'] = jail[3]
+                results['_meta']['hostvars'][iocage_name]['iocage_type'] = jail[4]
+                results['_meta']['hostvars'][iocage_name]['iocage_release'] = jail[5]
+                results['_meta']['hostvars'][iocage_name]['iocage_ip4'] = _parse_ip4(jail[6])
+                results['_meta']['hostvars'][iocage_name]['iocage_ip6'] = jail[7]
+                results['_meta']['hostvars'][iocage_name]['iocage_template'] = jail[8]
+                results['_meta']['hostvars'][iocage_name]['iocage_basejail'] = jail[9]
+
+        except Exception as e:
+            raise AnsibleParserError('Failed to parse %s: %s. iocage_data: %s' %
+                                     (to_native(path), to_native(e), to_native(iocage_data)))
```

Comment:
```suggestion
                                     (to_native(path), to_native(e), to_native(iocage_data))) from e
```
Also `iocage_data` is not necessarily defined at this point, for example if the above `AnsibleError`s have been raised.

Reply:
I removed iocage_data from the AnsibleParserError. Thank you!